# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1392224003 0
# Node ID ff888a8d8987c55c28ffab7e493e1fdf064696c2
# Parent  6fe8608c10fc1e7d879e9a6bca18f9b1648d1d64
libxc: Fix out-of-memory error handling in xc_cpupool_getinfo()

Avoid freeing info then returning it to the caller.

This is XSA-88.

Coverity-ID: 1056192
Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Reviewed-by: Jan Beulich <jbeulich@suse.com>

diff -r 6fe8608c10fc -r ff888a8d8987 tools/libxc/xc_cpupool.c
--- a/tools/libxc/xc_cpupool.c	Wed Feb 12 16:52:26 2014 +0000
+++ b/tools/libxc/xc_cpupool.c	Wed Feb 12 16:53:23 2014 +0000
@@ -104,6 +104,7 @@ xc_cpupoolinfo_t *xc_cpupool_getinfo(xc_
     info->cpumap = xc_cpumap_alloc(xch);
     if (!info->cpumap) {
         free(info);
+        info = NULL;
         goto out;
     }
     info->cpupool_id = sysctl.u.cpupool_op.cpupool_id;
